cloudformation/calculation-athena.yaml

```diff
@@ -179,14 +161,14 @@ Resources:
                         "Date_format"(timestamp, '%m') month,
                         "Date_format"(timestamp, '%Y') year
                 FROM (btm_transactions_standardised_with_count txn
-                        left join "btm_vendor_service_standardised" ven 
-                                ON ( txn.event_name = ven.event_name )
-                                  AND ( txn.vendor_id = ven.vendor_id )
-                        left join "standardised_rates_with_end" rates
-                                ON ( txn.event_name = rates.event_name )
-                                  AND ( txn.vendor_id = rates.vendor_id )
-                                  AND ( txn.month_event_count BETWEEN rates.volumes_from AND Coalesce(rates.volumes_to, 9999999))
-                                  AND ( txn.timestamp BETWEEN rates.effective_from AND rates."end" )))
+                        LEFT JOIN "btm_vendor_service_standardised" ven 
+                                ON (txn.event_name = ven.event_name)
+                                  AND (txn.vendor_id = ven.vendor_id)
+                        LEFT JOIN "btm_rates_standardised" rates
+                          ON (txn.event_name = rates.event_name)
+                            AND (txn.vendor_id = rates.vendor_id)
+                            AND (txn.month_event_count BETWEEN rates.volumes_from AND COALESCE(rates.volumes_to, 9999999))
+                            AND (txn.timestamp BETWEEN rates.effective_from AND COALESCE(rates.effective_to, "From_iso8601_timestamp"('2049-12-31')))))
```

Comment:
I did think of that; it would definitely feel more 'pure' that way, but it's an extra query, with a bit of complexity -- it would need to look at the max start date that's less than the event's start date, for _each_ event, as we're still returning the full set of events across months.

If in the future we query for just the current month (as I think we'll want to do at some point), I think that would be the time to put a change like that in...